### PR TITLE
UIOAIPMH-6: add filtering conditions for create/edit set

### PR DESCRIPTION
**Pull request by Dmitriy-Litvinenko**
# Purpose
Add filtering conditions for create/edit set
# Link
https://issues.folio.org/browse/UIOAIPMH-6
# Screenshot
image

### CODE REVIEW

File: test/bigtest/tests/sets/create-edit-test.js [hunk 1]
```diff
@@ -86,27 +89,129 @@ describe('Sets', () => {
...
+              .to.equal(newValues.value);
+          });
+
+          it('should change set spec', () => {
+            expect(SetsFormInteractor.filteringConditionsRow(0).setSpec.val)
+              .to.equal(newValues.setSpec);
+          });
+        });
+
+        describe('change values', () => {
```
Comment (ViktorSoroka07) [inline test/bigtest/tests/sets/create-edit-test.js:194]: Is it needed?

File: src/settings/routes/sets/SetsEditRoute.js [hunk 2]
```diff
@@ -162,6 +190,7 @@ SetsEditRoute.propTypes = {
   stripes: PropTypes.shape({
     hasPerm: PropTypes.func.isRequired,
   }).isRequired,
+  intl: PropTypes.object.isRequired,
 };
 
-export default stripesConnect(SetsEditRoute);
+export default injectIntl(stripesConnect(SetsEditRoute));
```
Comment (ViktorSoroka07) [inline src/settings/routes/sets/SetsEditRoute.js:196]: I would suggest starting using `useIntl` hook.

File: src/settings/routes/sets/SetsDuplicateRoute.js [hunk 3]
```diff
@@ -162,6 +186,7 @@ SetsDuplicateRoute.propTypes = {
   stripes: PropTypes.shape({
     hasPerm: PropTypes.func.isRequired,
   }).isRequired,
+  intl: PropTypes.object.isRequired,
 };
 
-export default stripesConnect(SetsDuplicateRoute);
+export default injectIntl(stripesConnect(SetsDuplicateRoute));
```
Comment (ViktorSoroka07) [inline src/settings/routes/sets/SetsDuplicateRoute.js:192]: `useIntl`

File: src/settings/routes/sets/SetsCreateRoute.js [hunk 4]
```diff
@@ -91,6 +117,7 @@ SetsCreateRoute.propTypes = {
   stripes: PropTypes.shape({
     hasPerm: PropTypes.func.isRequired,
   }).isRequired,
+  intl: PropTypes.object.isRequired,
 };
 
-export default stripesConnect(SetsCreateRoute);
+export default injectIntl(stripesConnect(SetsCreateRoute));
```
Comment (ViktorSoroka07) [inline src/settings/routes/sets/SetsCreateRoute.js:123]: `useIntl`

**Review by ViktorSoroka07 — APPROVED** [review]

File: test/bigtest/tests/sets/create-edit-test.js (hunk 1 above)
Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/sets/create-edit-test.js]: removed

File: src/settings/routes/sets/SetsEditRoute.js (hunk 2 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/routes/sets/SetsEditRoute.js]: fixed

File: src/settings/routes/sets/SetsCreateRoute.js (hunk 4 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/routes/sets/SetsCreateRoute.js]: fixed

File: src/settings/routes/sets/SetsDuplicateRoute.js (hunk 3 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/routes/sets/SetsDuplicateRoute.js]: fixed

File: src/settings/components/Sets/SetsForm/components/FilteringConditions/FilteringConditions.js [hunk 5]
```diff
@@ -1,23 +1,148 @@
...
 
-const FilteringConditions = () => {
+import {
+  FilteringConditionsName,
+  FilteringConditionsTitle,
+} from './components';
+
+import css from './FilteringConditions.css';
+
+const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.slice(1);
```
Comment (ryandberger) [inline src/settings/components/Sets/SetsForm/components/FilteringConditions/FilteringConditions.js:31]: Move this into /util/convertData.js

File: test/bigtest/tests/sets/create-edit-test.js [hunk 6]
```diff
@@ -161,14 +270,21 @@ describe('Sets', () => {
...
     };
     const currentPath = `/settings/oai-pmh/sets/${initialValues.id}/edit`;
     const redirectBackPath = `/settings/oai-pmh/sets/${initialValues.id}/view`;
 
-    beforeEach(function () {
+    beforeEach(async function () {
       this.server.create('set', initialValues);
 
       this.visit(currentPath);
+      await wait(300);
```
Comment (ryandberger) [inline test/bigtest/tests/sets/create-edit-test.js:287]: The preferred pattern to to await the presence of an expected element when the page is loaded. That way once the element is present, you can proceed which will likely be faster than 300ms. It is also more resilient if the load takes longer than 300ms so you don't need to do trial-and-error with arbitrary wait times.

Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/sets/create-edit-test.js]: fixed

File: src/settings/components/Sets/SetsForm/components/FilteringConditions/FilteringConditions.js (hunk 5 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/components/Sets/SetsForm/components/FilteringConditions/FilteringConditions.js]: moved

**Comment by Dmitriy-Litvinenko** [comment]
@ryandberger  Could you please re-review?

**Pull request merged**